client/src/components/NavBar/NavBar.tsx

```diff
@@ -24,7 +24,13 @@ const NavBar = (): JSX.Element => {
   }, [initSocket]);
 
   if (loggedInUser === undefined) return <CircularProgress />;
-  if (!loggedInUser && history.location.pathname !== '/login' && history.location.pathname !== '/signup') {
+  if (
+    !loggedInUser &&
+    loggedInUser === null &&
+    loggedInUser === undefined &&
```

Comment:
`!loggedInUser` covers all these three :smiley:

Reply:
Hi ezeilo-su, When I made these changes, It is giving me redirecting issues. Please watch this video 

https://share.vidyard.com/watch/iDaM7RnfbqiNNMAqCcV7oo?